pyannote/core/segment.py

```diff
@@ -65,15 +65,21 @@
 
 See :class:`pyannote.core.Segment` for the complete reference.
 """
+import warnings
+from typing import Union, Optional, Tuple, List, Iterator
 
-from collections import namedtuple
 import numpy as np
+from dataclasses import dataclass
 
 # 1 μs (one microsecond)
+from pyannote.core.utils.types import SegmentCropMode
```

Comment:
I am not a fan of name `SegmentCropMode`

I use `Alignment` in `pyannote.audio` v2: cf https://github.com/pyannote/pyannote-audio/blob/ace8534d70c21d7225453bed0eed526f12151efb/pyannote/audio/train/model.py#L48

Let's use this. I'll update pyannote.audio to use pyannote.core typing once this PR is merged and released.